components/browser/session/src/main/java/mozilla/components/browser/session/Session.kt

```diff
@@ -91,6 +91,10 @@ class Session(
             triggeredByRedirect: Boolean,
             triggeredByWebContent: Boolean
         ) = Unit
+        /**
+         * Indicates this session has had it's first engine contentful paint of page content.
+         */
+        fun onFirstContentfulPaint(session: Session) = Unit
```

Comment:
Is there any chance we can move this to `SessionState` and not add it to `Session`? Since we are moving away from `Session` I would love to not add anything new here if possible. :)

Reply:
Yep! We aren't using `TabSessionState` very much yet in Fenix I think? But something like this looks like it would work the same? 
https://github.com/mozilla-mobile/fenix/pull/10358/files#diff-bf0b57d467da0e3d5ef96c23fc828376R177

Reply:
Yeah, we could either subscribe to `BrowserStore` or lookup the `TabSessionState` if it is a one time thing.

Reply:
I just realized that I left my other comment in the issue yesterday and not here on the PR: https://github.com/mozilla-mobile/android-components/issues/6757#issuecomment-623379531 - If that is possible then maybe we can simplify it even more? 🤔

Reply:
I'm pretty sure Fenix still needs to be able to access whether the first contentful paint has happened for the case of navigate to browser -> click URL bar to enter SearchFragment -> Press Back. We are setting the EngineView to GONE while leaving the BrowserFragment for visual and performance issues, and need to set it to VISIBLE again (and we won't get the onFirstContentfulPaint callback for an existing session).

Reply:
Ah, yeah, that makes sense - we need to keep that state around. I still think we could perform the showing/hiding inside `GeckoEngineView` and show/hide the internal `GeckoView` based on whether the paint has happened or when it happens. Do you agree? (Of course this does not need to happen here in the PR).